src/transformStateWithClones.js

```diff
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          clone[key] = action.extraData[key];
+        };
+
+        result.push({ ...clone });
```

Comment:
you can move `result.push` after `switch` block to avoid repeated code

e.g.

```
switch {
  case ...
    break;
  case ...
    break;
}

push(...clone)
```

Reply:
Yes! You are right!